operators/src/upgrade/k8s/client.rs

```diff
@@ -0,0 +1,52 @@
+use crate::upgrade::k8s::crd::v0::UpgradeAction;
+use k8s_openapi::api::{core::v1::Node, storage::v1::StorageClass};
+use kube::{
+    api::{DeleteParams, ListParams},
+    core::ObjectList,
+    Api, Client,
+};
+
+use crate::upgrade::common::{constants::NODE_LABEL, error::Error};
+
+#[derive(Clone)]
+pub(crate) struct K8sClient {
+    client: kube::Client,
+}
+
+impl K8sClient {
+    /// Create a new K8sClient from default configuration.
+    pub(crate) async fn new() -> Result<Self, Error> {
+        let client = Client::try_default().await?;
+        Ok(Self { client })
+    }
+
+    pub(crate) fn client(&self) -> kube::Client {
+        self.client.clone()
+    }
+
+    /// Get nodes present in the cluster with mayastor labels present.
+    pub(crate) async fn get_nodes(&self) -> Result<ObjectList<Node>, Error> {
+        let nodes: Api<Node> = Api::all(self.client.clone());
+        let lp = ListParams::default().labels(NODE_LABEL);
+        let list = nodes.list(&lp).await?;
+        println!("{:?}", list);
+        Ok(list)
+    }
+
+    /// Get nodes present in the cluster with mayastor labels present.
+    pub(crate) async fn get_upgrade_action(&self) -> Result<Api<UpgradeAction>, Error> {
+        let uas: Api<UpgradeAction> = Api::all(self.client.clone());
+        Ok(uas)
+    }
+
+    /// Get nodes present in the cluster with mayastor labels present.
+    pub(crate) async fn delete_storage_class(&self) -> Result<(), Error> {
+        let nodes: Api<StorageClass> = Api::all(self.client.clone());
+        //let lp = ListParams::default().labels(NODE_LABEL);
+        nodes
+            .delete("mayastor-single-replica", &DeleteParams::default())
```

Comment:
You should not identify helm objects by their names, is there a unique label we can use instead which ties this storageclass to the helm release? If not, can we add a unique label to all resource templates in the helm chart which carries the `.Release.Name` value or equivalent?